Updating to the new sweet.js lib

SimonRichardson opened the pull request:
Also added `$kleisli`, ideally I would have liked to use infix operators, but this currently isn't supported by sweet.js. So I used the same idea as `$ap` for it.

SimonRichardson committed:
Updating to the new sweet.js lib
- Moved to a file based approach to make it easier to see what's happening
- Removed Mocha and using nodeunit inline with other fantasy-land packages.
- TODO - Implement do tests.

- Adding do tests.

- Removing bin from repo.

- Using `grunt` instead of `grunt macro`

SimonRichardson committed:
Adding readme and also updating package.json

markandrus commented:
Overall, +1! But regarding `$kleisli`, isn't this usually defined (at least in Haskell) as

``` hs
(>=>) :: Monad m => (a -> m b) -> (b -> m c) -> a -> m c
```

and so shouldn't

``` js
$kleisli (x, y, z) > a
```

desugar into

``` js
x(a).chain(y).chain(z)
```

rather than

``` js
a.chain(x).chain(y).chain(z)
```

?

Commit:
Fixing Kleisli macro.
- I got the arguments the wrong way

- Fixing semigroup ordering seems to break between different computers :-|

SimonRichardson commented:
@markandrus You are indeed correct, I did start off doing that way but got inspired by `$ap` and ended up breaking the original implementation. I've gone back to using `>=>` as we can't do what `$ap` is doing, which was using the `,` as a scalar repeat. In turn fixing the implementation. 

Glad you reviewed the code!

I wonder if using `>=>` as a repeating scalar should be allowed in sweet.js?
